cypress-io/cypress
# chore: move Linux binary packaging and publishing to separate repository

- Closes https://github.com/cypress-io/cypress-internal/issues/622

### Additional details


Right now we package and publish our production binary in our open source repo, allowing the logs of our CI processes to be publicly viewed. This increases our risk of exposing keys and secrets that should be hidden from the public. This process moves the packaging and publishing process to our Linux binaries (arm and amd) to a [separate repository](https://github.com/cypress-io/publish-binary) that is only accessibly internally. This way, the logs are not publicly accessible and we have an additional layer of protection against leaking private secrets.

This PR does the following:
- Breaks up "building" (`yarn binary-build`) and "packaging" into two separate tasks (`yarn binary-package`)
- Uses the Circle API to trigger a pipeline in the internal repo which packages the app from the built production assets
- Uses a Circle "approval" job in the Linux workflows to wait for internal pipeline to complete

See [this Whimsical](https://whimsical.com/private-ci-build-process-Mp6BeBqG4Sak46y6WUyhDD) that describes the process (internal only).

### Steps to test


The most important thing to test here is that everything works as it did before, especially in the Linux workflows. There should be no change with how the binary behaves. The only difference is that we moved the package and smoke testing processes somewhere else.

Pull down the binary for each OS and run it on your machine to make sure that everything works correctly.

### How has the user experience changed?


n/a

### PR Tasks


- [na] Have tests been added/updated?
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
